pkg/catalogv2/oci/client.go

```diff
@@ -0,0 +1,251 @@
+// parsing the URL provided.
+func NewOCIClient(url string) (*OCIClient, error) {
+	ociClient := &OCIClient{
+		URL: url,
+	}
+
+	err := ociClient.parseURL()
+	if err != nil {
+		return nil, err
+	}
+
+	return ociClient, nil
+}
+
+func (o *OCIClient) parseURL() error {
+	url := strings.TrimSuffix(o.URL, "/")
+	url = strings.TrimPrefix(url, "oci://")
+
+	if strings.Contains(url, "/") {
+		var ref registry.Reference
+		ref, err := registry.ParseReference(url)
+		if err != nil {
+			return fmt.Errorf("failed to parse OCI URL '%s' value: %w", o.URL, err)
+		}
+		o.registry = ref.Registry
+		o.repository = ref.Repository
+		o.tag = ref.Reference
+	} else {
+		o.registry = url
+	}
+
+	return nil
+}
+
+func (o *OCIClient) fetchChart(orasRepository *orasRegistryRemote.Repository) (string, bool, error) {
+
+	memoryStore := memory.New()
+	manifest, err := oras.Copy(context.Background(), orasRepository, o.tag, memoryStore, "", oras.DefaultCopyOptions)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to oras copy the remote oci artifact %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+	}
+
+	// Fetch the manifest content of the oci artifact
+	manifestblob, err := content.FetchAll(context.Background(), memoryStore, manifest)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to fetch the manifest blob of %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+	}
+
+	var manifestJson ocispec.Manifest
+	err = json.Unmarshal(manifestblob, &manifestJson)
+	if err != nil {
+		return "", false, fmt.Errorf("unable to unmarshal manifest blob of %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+	}
+
+	// Create a temp file to store the helm chart tar
+	tempFile, err := os.CreateTemp("", "helm-")
+	if err != nil {
+		return "", false, fmt.Errorf("unable to create temp file for storing the oci artifact")
+	}
+
+	// Check if the oci artifact is of type helm config ?
+	if manifest.ArtifactType == helmregistry.ConfigMediaType || manifestJson.Config.MediaType == helmregistry.ConfigMediaType {
+		// find the layer of chart type and fetch it
+		for _, layer := range manifestJson.Layers {
+			if layer.MediaType == helmregistry.ChartLayerMediaType {
+
+				chartTar, err := content.FetchAll(context.Background(), memoryStore, layer)
+				if err != nil {
+					return "", false, fmt.Errorf("unable to fetch chart blob of %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), err.Error())
+				}
+
+				err = os.WriteFile(tempFile.Name(), chartTar, 0644)
+				if err != nil {
+					return "", false, fmt.Errorf("unable to write chart %s into file %s:%s", fmt.Sprintf("%s/%s:%s", o.registry, o.repository, o.tag), tempFile.Name(), err.Error())
+				}
+
+				return tempFile.Name(), true, nil
+			}
+		}
+	}
+
+	return "", false, nil
+}
+
+func (o *OCIClient) getOrasRegistry(credentialSecret *corev1.Secret) (*orasRegistryRemote.Registry, error) {
```

Comment:
Please add a doc here describing the function

Reply:
Done.